.github/workflows/main.yml

```diff
@@ -56,24 +12,37 @@ jobs:
         # Github Actions requires a single row to be added to the build matrix.
         # See https://help.github.com/en/articles/workflow-syntax-for-github-actions.
         name: [
-          windows-clang-cl,
+          windows-clang,
         ]
         include:
-          - name: windows-clang-cl
-            os: windows-latest
+          - name: ubuntu-clang
+            os: ubuntu-16.04
+            compiler: clang
+          - name: windows-clang
+            os: windows-2019
             compiler: clang-cl
+          - name: macos-clang
+            os: macos-10.15
+            compiler: clang
     steps:
+    - name: Configure Git
+      shell: bash -e -o pipefail -l {0}
+      run: |
+        git config --global core.symlinks true && git config --global core.autocrlf false
+        if command -v dpkg > /dev/null; then sudo dpkg-reconfigure debconf -f noninteractive -p high; fi
     - name: Checkout repository
       uses: actions/checkout@v2
+      with:
+        fetch-depth: 1
     - name: Setup Bazel
-      shell: bash
+      shell: bash -e -o pipefail -l {0}
```

Comment:
this should be set in `install-bazel.sh` already, if not let's add it there?

Reply:
My goal here isn't really to add `-e -o pipefail`. Rather, I'm trying to add `-l` (because GitHub Actions doesn't source `~/.bashrc` etc. at all!), but in order to do that I have to add `-e -o pipefail` explicitly to match the flags GHA already uses by default (otherwise people will get confused).

Reply:
See here: https://github.community/t5/GitHub-Actions/How-to-share-shell-profile-between-steps-or-how-to-use-nvm-rvm/m-p/33633/highlight/true#M1587

Reply:
Or at least I think I do... unless it still passes those implicitly somehow. I can check if you'd like.

Reply:
I see. can you add a comment explain this choice in the first instantiation?

Reply:
Sure, will do!